src/components/app/details/triggerView/workflow.service.ts

```diff
@@ -80,13 +85,27 @@ export function processWorkflow(
                             return
                         }
                         wf.nodes.push(ciNode)
+                    } else if (branch.type == PipelineType.WEBHOOK) {
+                        let webhook = webhookMap.get(branch.componentId)
+                        if (!webhook) {
+                            return
+                        }
+                        let webhookNode = webhookToNode(webhook, dimensions)
+                        wf.nodes.push(webhookNode)
                     } else {
                         let cdPipeline = cdMap.get(branch.componentId)
                         if (!cdPipeline) {
                             return
                         }
                         let cdNode = cdPipelineToNode(cdPipeline, dimensions, branch.parentId)
                         let parentType = branch.parentType == PipelineType.CI_PIPELINE ? 'CI' : 'CD'
+                        if (branch.parentType == PipelineType.CI_PIPELINE) {
+                            parentType = 'CI'
+                        } else if (branch.parentType == PipelineType.WEBHOOK) {
+                            parentType = PipelineType.WEBHOOK
+                        } else {
+                            parentType = 'CD'
```

Comment:
Can we please remove else...if & else part? As we're already checking for CI/CD on Ln. 101.

Reply:
done